drivers/usb/device/usb_dc_nrf5.c

```diff
@@ -0,0 +1,2256 @@
+ */
+
+/** Number of IN Endpoints configured (including control) */
+#define CFG_EPIN_CNT (CONFIG_USBD_NRF5_NUM_IN_EP + \
+		      CONFIG_USBD_NRF5_NUM_BIDIR_EP)
+
+/** Number of OUT Endpoints configured (including control) */
+#define CFG_EPOUT_CNT (CONFIG_USBD_NRF5_NUM_OUT_EP + \
+		       CONFIG_USBD_NRF5_NUM_BIDIR_EP)
+
+/** Number of ISO IN Endpoints */
+#define CFG_EP_ISOIN_CNT CONFIG_USBD_NRF5_NUM_ISOIN_EP
+
+/** Number of ISO OUT Endpoints */
+#define CFG_EP_ISOOUT_CNT CONFIG_USBD_NRF5_NUM_ISOOUT_EP
+
+/** ISO endpoint index */
+#define EP_ISOIN_INDEX CFG_EPIN_CNT
+#define EP_ISOOUT_INDEX (CFG_EPIN_CNT + CFG_EP_ISOIN_CNT + CFG_EPOUT_CNT)
+
+/** Minimum endpoint buffer size */
+#define EP_BUF_MIN_SZ MAX_EP_BUF_SZ
+
+/** Maximum endpoint buffer size */
+#if (CFG_EP_ISOIN_CNT || CFG_EP_ISOOUT_CNT)
+#define EP_BUF_MAX_SZ MAX_ISO_EP_BUF_SZ
+#else
+#define EP_BUF_MAX_SZ MAX_EP_BUF_SZ
+#endif
+
+/** Total endpoints configured */
+#define CFG_EP_CNT (CFG_EPIN_CNT + CFG_EP_ISOIN_CNT + \
+		    CFG_EPOUT_CNT + CFG_EP_ISOOUT_CNT)
+
+/** Total buffer size for all endpoints */
+#define EP_BUF_TOTAL ((CFG_EPIN_CNT * MAX_EP_BUF_SZ) + \
+		      (CFG_EPOUT_CNT * MAX_EP_BUF_SZ) + \
+		      (CFG_EP_ISOIN_CNT * MAX_ISO_EP_BUF_SZ) +	\
+		      (CFG_EP_ISOOUT_CNT * MAX_ISO_EP_BUF_SZ))
+
+/** Total number of maximum sized buffers needed */
+#define EP_BUF_COUNT ((EP_BUF_TOTAL / EP_BUF_MAX_SZ) + \
+		      ((EP_BUF_TOTAL % EP_BUF_MAX_SZ) ? 1 : 0))
+
+/** 4 Byte Buffer alignment required by hardware */
+#define EP_BUF_ALIGN sizeof(unsigned int)
+
+K_MEM_POOL_DEFINE(ep_buf_pool, EP_BUF_MIN_SZ, EP_BUF_MAX_SZ,
+		  EP_BUF_COUNT, EP_BUF_ALIGN);
+
+/**
+ * @brief USBD private structure
+ *
+ * @param enabled	USBD Enable/Disable flag
+ * @param attached	USBD Attached flag
+ * @param ready		USBD Ready flag set after pullup
+ * @param address_set	USB Address set or unset
+ * @param state		USBD state
+ * @param status_code	Device Status code
+ * @param flags		Flags used in work context
+ * @param enable_mask	Enabled interrupts mask
+ * @param usb_work	USBD work item
+ * @param work_queue	FIFO used for queuing up events from ISR
+ * @param dma_in_use	Semaphore to restrict access to DMA one at a time
+ * @param status_cb	Status callback for USB DC notifications
+ * @param ep_ctx	Endpoint contexts
+ * @param buf		Buffer for the endpoints, aligned to 4 byte boundary
+ */
+struct nrf5_usbd_ctx {
+	bool enabled;
+	bool attached;
+	bool ready;
+	bool address_set;
+	enum nrf5_usbd_state state;
+	enum usb_dc_status_code status_code;
+	u32_t flags;
+	u32_t enable_mask;
+	struct k_work usb_work;
+	struct k_fifo work_queue;
+	struct k_sem dma_in_use;
+	usb_dc_status_callback status_cb;
+	struct nrf5_usbd_ep_ctx ep_ctx[CFG_EP_CNT];
+};
+
+static struct nrf5_usbd_ctx usbd_ctx;
+
+static inline struct nrf5_usbd_ctx *get_usbd_ctx(void)
+{
+	return &usbd_ctx;
+}
+
+static inline bool nrf5_usbd_ep_is_valid(const u8_t ep)
```

Comment:
In general for all static functions: remove all `nrf5_usbd_` prefix, this is not needed at all and makes the code too verbose

Reply:
Okay. Will do.